crates/moonbuild/src/new.rs

```diff
@@ -143,17 +143,21 @@ pub fn moon_new_lib(
 
 fn common(target_dir: &Path, cake_full_name: &str, license: Option<&str>) -> anyhow::Result<i32> {
     std::fs::create_dir_all(target_dir).context("failed to create target directory")?;
-    let git_init = std::process::Command::new("git")
-        .arg("init")
-        .current_dir(target_dir)
-        .status();
-    match git_init {
-        Ok(status) if status.success() => {}
-        _ => {
-            eprintln!(
-                "{}: git init failed, make sure you have git in PATH",
-                "Warning".yellow().bold()
-            );
+
+    let git_path = Path::new(".git");
```

Comment:
`target_dir.join(".git")`?

Reply:
Is the purpose of this PR to ensure that when `~/git/project` is a git repo, running `cd ~/git/project && moon new hello` or `cd ~/git/project/subdir && moon new hello` does not initialize a new git repo for the newly created project?

To achieve this, you need to check for the existence of a `.git` directory recursively or use another method to verify whether the current directory is within a git repo.

Reply:
Checking if `target_dir.join(".git")` exists is to prevent `git init` from being executed twice when running `moon new hello && moon new hello`. Although running `git init` twice is harmless.